lib/fuzz/fuzz_targets/fuzz_structured_exector.rs

```diff
@@ -0,0 +1,13 @@
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+use std::hint::black_box;
+use surrealdb::{dbs::Session, kvs::Datastore, sql::Query};
+
+fuzz_target!(|query: Query| {
+	tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap().block_on(async {
+		let dbs = Datastore::new("memory").await.unwrap();
+		let ses = Session::owner().with_ns("test").with_db("test");
+		_ = black_box(dbs.process(query, &ses, None).await);
```

Comment:
Not sure if this was the right approach, but I was operating off the documentation on the [process method](https://github.com/surrealdb/surrealdb/blob/main/lib/src/kvs/ds.rs#L1011) which says "Execute a pre-parsed SQL query". The `execute` URL you have linked points to a different function than the [execute function that is called here](https://github.com/surrealdb/surrealdb/blob/main/lib/src/kvs/ds.rs#L999). I chose not to use the `execute` method because it takes a [stringly typed query](https://github.com/surrealdb/surrealdb/blob/main/lib/src/kvs/ds.rs#L1001). Whereas the `process` function, just a few lines down does the same thing but on a preparsed query. In this case we aren't pre-parsing anything rather we are directly generating the query using arbitrary.

Reply:
Unless I misunderstood, the `execute` method [from the `dbs::Executor`](https://github.com/surrealdb/surrealdb/blob/main/lib/src/kvs/ds.rs#L1055) is the one called by `process` after checking for permissions and setting up the notification channel, as well as attaching any relevant variables and options. This `execute` method takes an AST. I don't think that is necessary for fuzzing, although I doubt it significantly affects performance in any way and it would also allow us to fuzz variable values at some point easily, we we might want to do.

I think it is fine to fuzz `process`, I just wanted to understand if there was a particular reason for the choice :+1: